dex/farm-staking-proxy/src/dual_yield_token.rs

```diff
@@ -0,0 +1,205 @@
+            ManagedAsyncCallResult::Err(err) => {
+                let (returned_tokens, token_id) = self.call_value().payment_token_pair();
+                if token_id.is_egld() && returned_tokens > 0 {
+                    let _ = self.send().direct_egld(caller, &returned_tokens, &[]);
+                }
+
+                OptionalResult::Some(err.err_msg)
+            }
+        }
+    }
+
+    fn require_dual_yield_token(&self, token_id: &TokenIdentifier) -> SCResult<()> {
+        let dual_yield_token_id = self.dual_yield_token_id().get();
+        require!(token_id == &dual_yield_token_id, "Invalid payment token");
+
+        Ok(())
+    }
+
+    fn require_all_payments_dual_yield_tokens(
+        &self,
+        payments: &ManagedVec<EsdtTokenPayment<Self::Api>>,
+    ) -> SCResult<()> {
+        if payments.is_empty() {
+            return Ok(());
+        }
+
+        let dual_yield_token_id = self.dual_yield_token_id().get();
+        for p in payments {
+            require!(
+                p.token_identifier == dual_yield_token_id,
+                "Invalid payment token"
+            );
+        }
+
+        Ok(())
+    }
+
+    fn create_and_send_dual_yield_tokens(
+        &self,
+        to: &ManagedAddress,
+        amount: &BigUint,
+        lp_farm_token_nonce: u64,
+        lp_farm_token_amount: BigUint,
+        staking_farm_token_nonce: u64,
+        staking_farm_token_amount: BigUint,
+    ) {
+        let dual_yield_token_id = self.dual_yield_token_id().get();
+        let empty_buffer = ManagedBuffer::new();
+        let attributes = DualYieldTokenAttributes {
+            lp_farm_token_nonce,
+            lp_farm_token_amount,
+            staking_farm_token_nonce,
+            staking_farm_token_amount,
+            total_dual_yield_tokens_for_position: amount.clone(),
+        };
+
+        let new_token_nonce = self.send().esdt_nft_create(
+            &dual_yield_token_id,
+            amount,
+            &empty_buffer,
+            &BigUint::zero(),
+            &empty_buffer,
+            &attributes,
+            &ManagedVec::new(),
+        );
+        self.send()
+            .direct(to, &dual_yield_token_id, new_token_nonce, amount, &[]);
+    }
+
+    fn burn_dual_yield_tokens(&self, sft_nonce: u64, amount: &BigUint) {
+        let dual_yield_token_id = self.dual_yield_token_id().get();
+        self.send()
+            .esdt_local_burn(&dual_yield_token_id, sft_nonce, amount);
+    }
+
+    fn get_dual_yield_token_attributes(
+        &self,
+        dual_yield_token_nonce: u64,
+    ) -> SCResult<DualYieldTokenAttributes<Self::Api>> {
+        let own_sc_address = self.blockchain().get_sc_address();
+        let dual_yield_token_id = self.dual_yield_token_id().get();
+        let token_info = self.blockchain().get_esdt_token_data(
+            &own_sc_address,
+            &dual_yield_token_id,
+            dual_yield_token_nonce,
+        );
+
+        token_info.decode_attributes().into()
```

Comment:
This method returns a `SCResult<DualYieldTokenAttributes>`. Should the return be `Ok(token_info.decode_attributes().into())` or remove the `SCResult` ?

Reply:
`decode_attributes` returns `Result<T, Err>`, so `into` converts it to a `SCResult` already.